src/main/java/com/aws/greengrass/lifecyclemanager/Kernel.java

```diff
@@ -603,15 +612,64 @@ public Kernel parseArgs(String... args) {
         }
 
         // Update device configuration from commandline arguments after loading config files
-        kernelCommandLine.updateDeviceConfiguration(getContext().get(DeviceConfiguration.class));
+        DeviceConfiguration deviceConfiguration = getContext().get(DeviceConfiguration.class);
+        kernelCommandLine.updateDeviceConfiguration(deviceConfiguration);
+        // After configuration is fully loaded, initialize Nucleus service config
+        initializeNucleusServiceConfigIfNotExist(kernelAlts, deviceConfiguration);
```

Comment:
There's some initialization logic in `DeviceConfiguration.initializeNucleusComponentConfig()` for initializing version and a few more things for the Nucleus component config, is there a way we can consolidate that with this here?

Reply:
I put it here because the interpolation should happen after all configs are loaded. Let me see if this can be optimized.

Reply:
I did some experiments. `initializeNucleusComponentConfig` was called in DeviceConfiguration constructor. Since I introduced ComponentStore RecipeLoader etc here in the init, changes in `initializeNucleusComponentConfig` will cause circular dependency issues. 

I still kept the logic here for now